galley/pkg/config/analysis/analyzers/auth/mtls/policy_checker.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mtls
+
+import (
+	"fmt"
+
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+	"istio.io/istio/security/proto/authentication/v1alpha1"
+)
+
+// Workload is a simple struct type for representing a service workload
+// targeted by an Authentication policy.
+type Workload struct {
+	fQDN string
+
+	// Either portNumber/portName will be set, or neither. The constructors
+	// prevent both from ever being set.
+	portNumber uint32
+	portName   string
```

Comment:
It's probably fine for an analyzer but canonical way to enforce this "union" kind of behavior is via interfaces in Golang i.e. you can have a struct element be an interface which underneath can be of different types with different storage.

Reply:
I thought about that, but decided it was too heavy-weight/complex for what's supposed to be a collection of helper methods (and I was reluctant to even make these members private!)

Good to know though, thanks!